core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java

```diff
@@ -223,12 +227,19 @@ public void stop() {
       // This excludes the keys from the various address targets
       Map<Address, Set<K>> keyTargets = determineKeyTargets(topology, keysToInclude, localAddress, segments, ctx);
 
+      int keyTargetSize = keyTargets.size();
+
+      if (keyTargetSize == 0) {
+         flowableProcessor.onComplete();
```

Comment:
I think just calling `onComplete()` is maybe not enough. Because, if the `transformer` function is not part of the (empty) pipeline then (for some cases) no "identity" element is returned finally.

The cases where this will happen is when this method of `DistributedCacheStream` is used:
```
   public R reduce(R identity, BinaryOperator<R> accumulator) {
      return performPublisherOperation(PublisherReducers.reduce(identity, accumulator),
            PublisherReducers.reduce(accumulator));
   }
```
Here only the `transformer` but not the `finalizer` takes care of the "identity" element.
I can create a unit-test to show the point if you want.

Reply:
I understand what you are saying and it is a good point, This is important for other Flowable mappings such as `Flowable.switchIfEmpty` etc. I was being a bit overzealous on the code reductions and your prior PR is probably the one we will want to use.

Reply:
But I may have something else still. Either way I will a couple more test cases, unless you have one available on you I can copy ;)

Reply:
Sure, what do you think about this testcase?
```
   @Test(dataProvider = "GuaranteeParallelEntry")
   public void testReduceTransformerOnlyIdentityEmptyKeys(DeliveryGuarantee deliveryGuarantee, boolean parallel, boolean isEntry) {
      Cache<Integer, String> cache = cache(0);
      int insertAmount = insert(cache).size();
      assertTrue(insertAmount > 0);

      // This empty set filters everything, but the computation must still complete
      Set<Integer> keysToInclude = new HashSet<>();

      // The identity element must still be returned, even if only the transformer (but not the finalizer) function takes care of it
      ClusterPublisherManager<Integer, String> cpm = cpm(cache);
      Object identity;
      CompletionStage<?> stage;
      if (isEntry) {
         SerializableBinaryOperator<CacheEntry<Integer, String>> binOp = (e1, e2) -> e1;
         identity = NullCacheEntry.getInstance();
         stage = cpm.entryReduction(parallel, null, keysToInclude, null, false, deliveryGuarantee,
               PublisherReducers.reduce((CacheEntry<Integer, String>) identity, binOp), PublisherReducers.reduce(binOp));
      } else {
         SerializableBinaryOperator<Integer> binOp = (k1, k2) -> k1;
         identity = 0;
         stage = cpm.keyReduction(parallel, null, keysToInclude, null, false, deliveryGuarantee,
               PublisherReducers.reduce((Integer) identity, binOp), PublisherReducers.reduce(binOp));
      }

      Object actual = stage.toCompletableFuture().join();
      Object expected = identity;
      assertTrue(expected == actual);
   }
```

Reply:
Okay, I was able to fix the reduction issue. However, looking at Publisher there is a different issue that I am not sure if I can resolve or not. I will update later though.